lib/ansible/modules/cloud/amazon/dynamodb.py

```diff
@@ -0,0 +1,390 @@
+                                          ExpressionAttributeValues=expression_attribute_values)
+    else:
+        response = connection.update_item(
+            TableName=table, Key=primary_key, UpdateExpression=update_expression)
+
+    changed = True
+    return response, changed
+
+
+def delete(connection, table, primary_key, condition_expression,
+           expression_attribute_values):
+
+    response = connection.delete_item(TableName=table, Key=primary_key, ConditionExpression=condition_expression,
+                                      ExpressionAttributeValues=expression_attribute_values)
+
+    changed = True
+    return response, changed
+
+
+def main():
+    argument_spec = dict(
+        table=dict(required=True),
+        action=dict(required=True, choices=["get", "put", "update", "delete"]),
+        primary_key=dict(required=False, type='dict'),
+        filter_expression=dict(required=False, type='str'),
+        item=dict(required=False, type='dict'),
+        condition_expression=dict(required=False, type='str'),
+        expression_attribute_names=dict(required=False, type='dict'),
+        expression_attribute_values=dict(required=False, type='dict'),
+        update_expression=dict(required=False, type='str')
+    )
+
+    module = AnsibleAWSModule(argument_spec=argument_spec,
+                              supports_check_mode=False)
+
+    connection = module.client('dynamodb')
+
+    try:
+        table = module.params.get('table')
+        action = module.params.get('action')
+        primary_key = module.params.get('primary_key')
+        filter_expression = module.params.get('filter_expression')
+        condition_expression = module.params.get('condition_expression')
+        item = module.params.get('item')
+        update_expression = module.params.get('update_expression')
+        expression_attribute_names = module.params.get(
+            'expression_attribute_names')
+        expression_attribute_values = module.params.get(
+            'expression_attribute_values')
+
+        changed = False
+        response = {}
+
+        if action == 'get':
+            response, changed = get(
+                connection, table, filter_expression,
+                expression_attribute_names, expression_attribute_values)
+
+        elif action == 'put':
+            response, changed = put(connection, table, item)
+
+        elif action == 'update':
+            response, changed = update(connection, table, primary_key, update_expression,
+                                       condition_expression, expression_attribute_names,
+                                       expression_attribute_values)
+
+        elif action == 'delete':
+            response, changed = delete(connection, table, primary_key,
+                                       condition_expression, expression_attribute_values)
+
+    except botocore.exceptions.ClientError as e:
```

Comment:
exception handling is better done in the individual methods so that more informative error messages can be returned (e.g. "could not update item")

Reply:
but maybe different methods will trigger same errors. i put it like that to save duplicated code.

Reply:
I understand that, but you can give more precise error messages.

Either way, you need to catch BotoCoreError, surely?

Reply:
this one is here isnt it ?https://github.com/blastik/ansible/blob/fa252990bb83b714215d90cf73dd37dcc203aed9/lib/ansible/modules/cloud/amazon/dynamodb.py#L386